### DISCUSSION TITLE
Github Actions / CI

### DISCUSSION THREAD

**Discussion by tritao** [nodejs-mobile/nodejs-mobile#59] [Ideas]
Hey, I was wondering why there is not a Github Actions setup that could build the project and upload new releases automatically.

Seems like it would be pretty useful given all the configurations available.

Is this just because it's not been setup yet?

**Comment by staltz** [comment]
You could try, but it stretches the limits of any system. Takes hours to compile the whole system, needs mobile SDKs not usually available in the cloud, and requires a lot of RAM. I wouldn't put my hopes up.

If I remember correctly, the iOS builds also require 100GB of disk space.

**Comment by tritao** [comment]
> You could try, but it stretches the limits of any system. Takes hours to compile the whole system, needs mobile SDKs not usually available in the cloud, and requires a lot of RAM. I wouldn't put my hopes up.

Ok, I put a quick script up for Android on Linux to see how it goes: https://github.com/tritao/nodejs-mobile/actions/runs/6676199802/job/18145019005

If it works, we can set up sccache too which should really speed up sucessive builds.

> If I remember correctly, the iOS builds also require 100GB of disk space.

Damn, that seems like a lot... If Actions manages to build Android, I'll give iOS a try to see how it goes.

**Comment by tritao** [comment]
Android builds in less than 2 hours, without any caching, which seems pretty reasonable, all things considered.

**Comment by tritao** [comment]
https://github.com/nodejs-mobile/nodejs-mobile/pull/60
